packages/plugin-kit/src/fs/writeProjectEnvironment.ts

```diff
@@ -0,0 +1,62 @@
+import {
+	checkHasProjectFile,
+	CheckHasProjectFileArgs,
+} from "./checkHasProjectFile";
+import { readProjectFile, ReadProjectFileArgs } from "./readProjectFile";
+import { writeProjectFile, WriteProjectFileArgs } from "./writeProjectFile";
+
+export type WriteProjectEnvironmentArgs = {
+	variableName: string;
+	environment: string | undefined;
+} & Omit<CheckHasProjectFileArgs, "filename"> &
+	Omit<ReadProjectFileArgs, "filename"> &
+	Omit<WriteProjectFileArgs, "contents" | "format" | "formatOptions">;
+
+export const writeProjectEnvironment = async (
+	args: WriteProjectEnvironmentArgs,
+): Promise<string | undefined> => {
+	const hasEnvironmentVariableFile = await checkHasProjectFile({
+		filename: args.filename,
+		helpers: args.helpers,
+	});
+
+	if (!hasEnvironmentVariableFile && args.environment === undefined) {
+		return;
+	}
+
+	let contents = hasEnvironmentVariableFile
+		? await readProjectFile({
+				filename: args.filename,
+				helpers: args.helpers,
+				encoding: "utf8",
+		  })
+		: "";
+
+	const hasExistingVariable = new RegExp(`^${args.variableName}=.*$`, "m").test(
+		contents,
+	);
```

Comment:
Would it be "safer" to use `dotenv.parse` instead here?

Not sure since there's no real workaround for after 🤔 Too bad `dotenv.stringify` does not exists.

Reply:
I figured we could get away with a RegExp, but you're right. `dotenv` would be safer, even if the likelihood of the RegExp failing is low.

I switched the RegExp for `dotenv.parse`.

We wouldn't want to do a `dotenv.parse` then `dotenv.print` (if it existed) because we would probably lose whitespace and comments. It's the same problem with parsing and printing a TypeScript file with `tsc`.

Reply:
Oh, right, great then!